rules/directive-restrict.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+module.exports = function(context) {
+    var utils = require('./utils/utils');
+
+    var options = context.options[0] || {};
+    var restrictOpt = options.restrict || 'AE';
+    var explicitRestrict = options.explicit === 'always';
+    var restrictChars = [];
+
+    restrictOpt.split('').forEach(function(char) {
+        if ('ACEM'.indexOf(char) === -1) {
```

Comment:
Ok, I understand your intention, but these lines of code only filter all chars not in 'ACEM' what is already checked by the regex, so line 13 can not be executed ever and that's why the line can not be covered by any test.

BUT I think your ACA corner case should be checked. And it can be checked with a regex.

/^A?C?E?M?$/

https://www.debuggex.com/r/nJTGB4V9fYnABGKt